File: DBProject5.php
<?php 
function execute($arg) {
	$command = 'java -cp .:mysql-connector-java-5.1.40-bin.jar DBProject5 ';
	echo $command;
	for ($i = 0; $i < count($arg); $i++) {
		$arg[$i] = escapeshellarg($arg[$i]);
		$command = $command . $arg[$i] . ' ';
	}
		
	$command = escapeshellcmd($command);
	echo "<p>command: $command </p>";
	system($command);
}


if (isset($_REQUEST['choice'])) {
	$choice = $_REQUEST['choice'];
	if ($choice == 1) { ?>
		<form action="DBProject5.php?choice=1" method="post">
			Add a Student to the student table<br>
		    Student Name: <input type="text" name="name"><br>
		    Student ID: <input type="text" name="ID"><br>
		    Major: <input type="text" name="major"><br>
		    <input name="submit" type="submit" >
		</form>
		<br><br>
		<?php
		if (isset($_POST['submit'])) {
			$input = array($_REQUEST['choice'], $_REQUEST['name'], $_REQUEST['ID'], $_REQUEST['major']);
	 		execute($input);          
		}
	}
	
	elseif ($choice == 2) { ?>
		<form action="DBProject5.php?choice=2" method="post">
			Add a Course to the Course Table<br>
		    Department Code: <input type="text" name="code"><br>
		    Course Number: <input type="text" name="courseNo"><br>
		    Course Title: <input type="text" name="title"><br>
		    Credit Hours: <input type="text" name="hours"><br>
		    <input name="submit" type="submit" >  
		</form>
		<br><br> 
		<?php
		if (isset($_POST['submit'])) {
			$input = array($_REQUEST['choice'], $_REQUEST['code'], $_REQUEST['courseNo'], $_REQUEST['title'], $_REQUEST['hours']);
	 		execute($input);          
		}
	}
	
	elseif ($choice == 3) { ?>
		<form action="DBProject5.php?choice=3" method="post">
			Enrollment<br>
		    Enter Student ID: <input type="text" name="studentID"><br>
		    Enter Department Code: <input type="text" name="deptCode"><br>
		    Course Number: <input type="text" name="courseNumber"><br>
		    <input name="submit" type="submit" >  
		</form>
		<br><br> 
		<?php
		if (isset($_POST['submit'])) {
			$input = array($_REQUEST['choice'], $_REQUEST['studentID'], $_REQUEST['deptCode'], $_REQUEST['courseNumber']);
	 		execute($input);          
		}
	}
	
	elseif ($choice == 4) {
		$input = array($_REQUEST['choice']);
	 	execute($input);          
	}
	
	elseif ($choice == 5) { ?>
		<form action="DBProject5.php?choice=5" method="post">
			Enter which department you would like to view the courses from:<br>
			<input type="text" name="deptCode">
			<input name="submit" type="submit" >  
			</form>
		<br><br> 
		<?php
		if (isset($_POST['submit'])) {
			$input = array($_REQUEST['choice'], $_REQUEST['deptCode']);
	 		execute($input);          
		}
	}
	
	elseif ($choice == 6) { ?>
		<form action="DBProject5.php?choice=6" method="post">
			Enter the ID of the student you would like to learn more about:<br>
			<input type="text" name="studentID">
			<input name="submit" type="submit" >  
			</form>
		<br><br>
		<?php
		if (isset($_POST['submit'])) {
			$input = array($_REQUEST['choice'], $_REQUEST['studentID']);
	 		execute($input);          
		}
	} 
}
?>

File: DBProject5.java
import java.sql.*;

/*
jdbc_insert_restaurant.java    // java program that is called by php that just does the insert; calls jdbc_db.java to connect and do the actual insert
jdbc_db.java // class (no main program) that has useful methods
*/

public class DBProject5
{
   // The main program that inserts a restaurant
   public static void main(String[] args) throws SQLException 
   {
		String Username = "arutiaga";              // Change to your own username
		String mysqlPassword = "LaLaLand";    // Change to your own mysql Password
		int choice = Integer.parseInt(args[0]); // Menu Item	
		// Connect to the database
		jdbc_db myDB = new jdbc_db();	
		myDB.connect(Username, mysqlPassword);			    
		// For debugging purposes:  Show the database before the insert
		StringBuilder builder = new StringBuilder();
		
		
		if(choice == 1)
		{	
			String query = "SELECT * FROM Student";
			builder.append("<br> Table Student before:" + myDB.query(query) + "<br>"); 
			
			String studentName = args[1];
			String studentID = args[2];
			String Major = args[3];

			// Insert the new restaurant
			String input = "" + studentID + ", '" + studentName + "', '" + Major + "'";
			System.out.println(input);              
			myDB.insert("Student", input);    // insert new student
			
					// For debugging purposes:  Show the database after the insert
			builder.append("<br><br><br> Table Student after:" + myDB.query(query));
			System.out.println(builder.toString()); 
			
		}
		
		else if(choice == 2)
		{
			String query = "SELECT * from Course";
			builder.append("<br> Table Course before:" + myDB.query(query) + "<br>"); 
			
			String departmentCode = args[1];
			String courseNumber = args[2];
			String courseTitle = args[3];
			String creditHours = args[4];
			
			String input = "'" + departmentCode + "'," + courseNumber + ",'" + courseTitle
							+ "'," + creditHours;
							
			myDB.insert("Course", input);
			builder.append("<br><br><br> Table Student after:" + myDB.query(query));
			System.out.println(builder.toString()); 
		}
		
		else if(choice == 3)
		{
			String query = "SELECT * from Enrollment";
			builder.append("<br> Table Enrollment before:" + myDB.query(query) + "<br>"); 
			
			String studentID = args[1];
			String departmentCode = args[2];
			String courseNumber = args[3];
			
			String input = "" + studentID + ",'" + departmentCode + "'," + courseNumber + "";
			myDB.insert("Enrollment", input);
			
			builder.append("<br><br><br> Enrollment Table after:" + myDB.query(query));
			System.out.println(builder.toString()); 
		}
		
		else if(choice == 4)
		{
			String query = "SELECT * FROM Student";
			builder.append("<br> All students:" + myDB.query(query) + "<br>");
			System.out.println(builder.toString()); 
		}
		
		else if(choice == 5)
		{
			String department = args[1];
			String query = "SELECT * " +
				    	   "FROM Course " +
				    	   "WHERE deptCode='" + department + "'";
			builder.append("<br> All courses from the " + department + " department: " + myDB.query(query) + "<br>");
			System.out.println(builder.toString()); 
		}
		
		else if(choice == 6)
		{
			Integer studentID = Integer.parseInt(args[1]);
			String query = "SELECT c.deptCode, c.courseNum, c.title, c.creditHours " +
						   "FROM Enrollment e, Course c " +
						   "WHERE e.deptCode=c.deptCode " +
						   "AND e.courseNum=c.courseNum " +
						   "AND e.studentID=" + studentID.toString();
			builder.append("<br> All courses that student " + studentID + " is taking: " + myDB.query(query) + "<br>");
			System.out.println(builder.toString()); 
		}		    

		myDB.disConnect();
	}
}
